c2corg_ui/static/js/outingscsvdownload.js

```diff
@@ -0,0 +1,150 @@
+goog.provide('app.OutingsCsvDownloadController');
+goog.provide('app.outingsCsvDownloadDirective');
+
+goog.require('app');
+
+
+/**
+ * This directive is used to display a download button for outings in CSV format.
+ *
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.outingsCsvDownloadDirective = function() {
+  return {
+    restrict: 'E',
+    controller: 'AppOutingsCsvDownloadController',
+    controllerAs: 'csvCtrl',
+    templateUrl: '/static/partials/outingscsvdownload.html'
+  };
+};
+
+app.module.directive('appOutingsCsvDownload', app.outingsCsvDownloadDirective);
+
+
+/**
+ * @param {app.Authentication} appAuthentication
+ * @param {app.Api} appApi The API service
+ * @param {angular.$q} $q Angular q service.
+ * @param {app.Lang} appLang Lang service.
+ * @constructor
+ * @struct
+ * @ngInject
+ */
+app.OutingsCsvDownloadController = function(appAuthentication, appApi, $q,
+    appLang) {
+
+  /**
+   * @type {app.Authentication}
+   * @export
+   */
+  this.auth = appAuthentication;
+
+  /**
+   * @type {app.Api}
+   * @export
+   */
+  this.appApi = appApi;
+
+  /**
+   * @type {angular.$q}
+   * @private
+   */
+  this.q_ = $q;
+
+  /**
+   * @type {app.Lang}
+   * @private
+   */
+  this.langService_ = appLang;
+};
+
+
+/**
+ * @param {goog.events.Event | jQuery.Event} event
+ * @export
+ */
+app.OutingsCsvDownloadController.prototype.download = function(event) {
+  event.stopPropagation();
+  var filename = 'outings.csv';
+  var langService = this.langService_;
+  this.getOutings_().then(function(outings) {
+    var csvFile = '';
+    outings.forEach(function(outing) {
+      var cells = [
+        outing.date_start,
+        outing.locales[0].title,
+        outing.activities.map(function(activity) {
+          return langService.translate(activity);
+        }).join(';'),
+        'https://www.camptocamp.org/outings/' + outing.document_id
+      ];
+      csvFile += '"' + cells.map(function(cell) {
+        if (cell.indexOf('"') > -1) {
+          return cells.replace(/"/g,'""');
```

Comment:
There is probably a typo here: ``cell`` instead of ``cells``.
In the demo the CSV export fails with error:
```
TypeError: cells.replace is not a function
    at http://localhost:6555/static/0dd7605/js/outingscsvdownload.js:84:24
```